core/src/main/java/org/kiwix/kiwixmobile/core/search/DefaultAdapter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search;
+
+import android.content.Context;
+import android.text.Html;
+import android.text.style.ClickableSpan;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+  private Context context;
+  List<String> searchList;
```

Comment:
this is not initialised so it is null, I believe this crashes right now

Reply:
```
2575
01:14:39 V/InstrumentationResultParser: There was 1 failure:
2576
01:14:39 V/InstrumentationResultParser: 1) SearchActivitySimple(org.kiwix.kiwixmobile.search.SearchActivityTest)
2577
01:14:39 V/InstrumentationResultParser: Test instrumentation process crashed. Check org.kiwix.kiwixmobile.search.SearchActivityTest#SearchActivitySimple.txt for details
```
[Yup](https://travis-ci.com/kiwix/kiwix-android/builds/139140868)

Reply:
```
  public void addAll(List<String> recentSearches) {
    searchList.addAll(recentSearches);
  }
```

Is it fine? Still, it is not being initialied but we are adding elements to it when addRecentSearches is called.

Reply:
You can't call methods on a null object, this is the reason for Null Pointer Exceptions.

Perhaps your confusion is arising from how `primitives` work, say int/boolean
```
int aDefaultValueInt; //same as writing int aDefaultValueInt = 0;
```

this only is true for the primitives, every other object is initialized to null

```
Integer aDefaultValueInteger; //same as writing Integer aDefaultValueInteger = null;
```
*Note the `boxed type` in the example above.

This is because when no value is specified java supplies a default value, primitives can not be null so it is `false` or `0` but the logical default value for an Object is `null`.

Initialise the field at the declaration or in the constructor so it can be marked final.